### ISSUE TITLE
Python 3 compatibility

### ISSUE THREAD

**Issue by Psykopear** [OpenBazaar/OpenBazaar#1336]
Why not make the whole project python3 compatible using future compatibility package?

**Comment by hoffmabc** [comment]
Can we do this without impacting dependencies like Twisted?

**Comment by Psykopear** [comment]
Yep, using __future__ imports the code is still interpreted by python2, but it's easier to migrate to python3 when every dependency has been ported

**Comment by Renelvon** [comment]
@Psykopear Using `__future__` would be a nice step towards Python3 compatibility. If you'd like, submit a PR and I will review it. Apart from that, we won't consider moving to Python 3 until all our dependencies have done so as well. Of course, we are always trying to write code that is compatible with Python 3; we have multiple relevant `pylint` checks in place.

**Comment by Psykopear** [comment]
Ok, i jumped into the code this afternoon, looking to understand how everything works, but i guess it will take a while.  I  noticed that the code is a bit messy. Some parts are clean and tested, but the most of the codebase is not. I guess to add __future__ and fix all the pep8 problems shouldn't take more than few hours, i'm used to do this kind of  refactoring on tons of lines of code just to be able to read everything better, but not knowing what every function does, i don't feel comfortable in "touching" untested code. That said, i'm really interested in the project, and planning to spend a part of my time in it, but i think i will give high priority to try to write unit tests to cover most of the codebase and learn how it works along the way. Where do people on this project talks to each other other than here? I'll probably have some silly question to ask at the beginning, and it would be great to talk with someone who has the know-how to give me a general vision of the flow of this project. I hope i'll be able to push some usefull code asap.

**Comment by hoffmabc** [comment]
Send your email address to project@openbazaar.org and someone will invite you to our slack.

**Comment by ShepTanner** [comment]
As a developer... I know the following question will not likely be initially welcomed.  
Has anyone considered rust?

**Comment by gubatron** [comment]
I think rust is a hard sell atm with no bitcoin libraries, did a quick google and rust-bitcoin is incomplete and there's only 2 guys working on it. Rust barely made it out of beta recently, I guess it's a bit too rushed to consider it at the moment (as cool as it sounds when you read the docs about this language)
